math/vecops/inc/ROOT/RVec.hxx

```diff
@@ -94,32 +111,985 @@ auto MapFromTuple(Tuple_t &&t, std::index_sequence<Is...>)
+
+   static void destroy_range(T *S, T *E)
+   {
+      while (S != E) {
+         --E;
+         E->~T();
+      }
+   }
+
+   /// Move the range [I, E) into the uninitialized memory starting with "Dest",
+   /// constructing elements as needed.
+   template <typename It1, typename It2>
+   static void uninitialized_move(It1 I, It1 E, It2 Dest)
+   {
+      std::uninitialized_copy(std::make_move_iterator(I), std::make_move_iterator(E), Dest);
+   }
+
+   /// Copy the range [I, E) onto the uninitialized memory starting with "Dest",
+   /// constructing elements as needed.
+   template <typename It1, typename It2>
+   static void uninitialized_copy(It1 I, It1 E, It2 Dest)
+   {
+      std::uninitialized_copy(I, E, Dest);
+   }
+
+   /// Grow the allocated memory (without initializing new elements), doubling
+   /// the size of the allocated memory. Guarantees space for at least one more
+   /// element, or MinSize more elements if specified.
+   void grow(size_t MinSize = 0);
+
+public:
+   void push_back(const T &Elt)
+   {
+      if (R__unlikely(this->size() >= this->capacity()))
+         this->grow();
+      ::new ((void *)this->end()) T(Elt);
+      this->set_size(this->size() + 1);
+   }
+
+   void push_back(T &&Elt)
+   {
+      if (R__unlikely(this->size() >= this->capacity()))
+         this->grow();
+      ::new ((void *)this->end()) T(::std::move(Elt));
+      this->set_size(this->size() + 1);
+   }
+
+   void pop_back()
+   {
+      this->set_size(this->size() - 1);
+      this->end()->~T();
+   }
+};
+
+// Define this out-of-line to dissuade the C++ compiler from inlining it.
+template <typename T, bool TriviallyCopyable>
+void SmallVectorTemplateBase<T, TriviallyCopyable>::grow(size_t MinSize)
+{
+   // Ensure we can fit the new capacity.
+   // This is only going to be applicable when the capacity is 32 bit.
+   if (MinSize > this->SizeTypeMax())
+      this->report_size_overflow(MinSize);
+
+   // Ensure we can meet the guarantee of space for at least one more element.
+   // The above check alone will not catch the case where grow is called with a
+   // default MinSize of 0, but the current capacity cannot be increased.
+   // This is only going to be applicable when the capacity is 32 bit.
+   if (this->capacity() == this->SizeTypeMax())
+      this->report_at_maximum_capacity();
+
+   // Always grow, even from zero.
+   size_t NewCapacity = size_t(NextPowerOf2(this->capacity() + 2));
+   NewCapacity = std::min(std::max(NewCapacity, MinSize), this->SizeTypeMax());
+   T *NewElts = static_cast<T *>(malloc(NewCapacity * sizeof(T)));
+
+   // Move the elements over.
+   this->uninitialized_move(this->begin(), this->end(), NewElts);
+
+   if (this->Owns()) {
+      // Destroy the original elements.
+      destroy_range(this->begin(), this->end());
+
+      // If this wasn't grown from the inline copy, deallocate the old space.
+      if (!this->isSmall())
+         free(this->begin());
```

Comment:
```suggestion
         std::free(this->begin());
```
But that is certainly debatable. There are more occurences of this in this PR.

Reply:
this is how ROOT calls it :man_shrugging: